modules/setting/hooks.go

```diff
@@ -0,0 +1,42 @@
+// Copyright 2024 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package setting
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"code.gitea.io/gitea/modules/log"
+)
+
+var (
+	// Git hook settings
+	GitHookPrereceiveName  string
+	GitHookPostreceiveName string
+	GitHookUpdateName      string
+)
+
+func isValidFileName(filename string) error {
+	if filepath.Base(filename) != filename || filepath.IsAbs(filename) || filename == "." || filename == ".." {
+		return fmt.Errorf("can only contain filenames, not other directories")
+	}
+	return nil
+}
```

Comment:
There is already a very similar function in `routers/api/packages/generic/generic.go:65`, move it to `modules/util` and share from there.